File: BootCamp Java/Car/Point.java
public class Point {
    private double x;
    private double y;
    
    public Point(double x, double y) {
        this.x = x;
        this.y = y;
    }
    public Point() {
        this(0, 0);
    }
    public double getX()  {
        return this.x;
    }

    public double getY()  {
        return this.y;
    }


    public double distance( Point other)  {
        return Math.hypot(this.x - other.x, this.y - other.y);
    }

    public boolean isEqual( Point other) {
        return this.x == other.x && this.y == other.y;
    }

    public boolean isNotEqual( Point other)  {
        return !isEqual(other);
    }
    public void pointPrint() {
        System.out.println("("+getX()+","+getY()+")");
    }
}

File: BootCampOOP/Basic/Point/Point.cpp
#include "Point.h"


Point::Point(double xIn, double yIn) : x(xIn), y(yIn) {

}

Point::~Point() {}

double Point::getX() const{
    return x;
}

double Point::getY() const{
    return y;
}

void Point::setX(double value){
    x = value;
}

void Point::setY(double value){
    y = value;
}

double Point::distance(const Point& other) const {
    
    return hypot(other.x-x, other.y-y);
    
}

bool Point::operator==(const Point& other) const {
    return (x == other.getX() && y == other.getY());
}

bool Point::operator!=(const Point& other) const {
    return !operator==(other);
}

std::ostream& operator<<(std::ostream& out, const Point& point) {
    out << "(" << point.getX() << ", " << point.getY() << ")";
    return out;
}